interpreter/CMakeLists.txt

```diff
@@ -131,6 +131,10 @@ else()
   set(CMAKE_CXX_VISIBILITY_PRESET hidden)
   set(CMAKE_C_VISIBILITY_PRESET hidden)
 endif()
+if (CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
+  ROOT_ADD_CXX_FLAG(CMAKE_CXX_FLAGS "-fno-semantic-interposition")
+endif()
```

Comment:
We likely also want this for Clang, right? (or is `CMAKE_CXX_COMPILER_ID` set to `GNU` in that case as well?)

Reply:
I found this restriction in https://reviews.llvm.org/D102453:

> Note: we cannot add -fno-semantic-interposition for Clang. Clang's implementation additionally optimizes global variables, which is incompatible with unfortunate ELF -fno-pic default: direct access relocations for external data. If the executable has a -fno-pic object file referencing a global variable declared in a public header, the direct access relocation will cause a copy relocation. The executable and libLLVM.so/libclang-cpp.so will disagree on the address.

OTOH why not try :-)

Reply:
Ugh, unfortunate. On the other hand, we don't want users to call into LLVM directly so we should be fine?